### PR TITLE
Server Re-write

### PR DESCRIPTION
**Pull request by sauravhiremath**
- [x] Server setup
- [x] Schema added
- [x] Controllers added
- [x] GitHub OAuth2
- [x] Discord OAuth2
- <s>Google OAuth2</s> (pushed for further release, due to low priority)

## How to check
- Add the env variables (discord and github client keys)
- Go to `/server`
- Run `yarn install`
- `yarn start`
- For Github login
  - Open `http://localhost:5000/auth/github/login`
    - It should ask you your acc permissions
    - It returns success with access-token and your profile details
    - Store it with react in localStorage/Cookie idc
- For discord login
  - Open `http://localhost:5000/auth/discord/login`
  - Should throw error as github-access-token not passed (meaning user not logged in)
  - Try again passing with `http://localhost:5000/auth/discord/login?token=abcdefg`
  - Give access to discord
  - Should return an object with discord and github details combined

## Env variables sample

```
NODE_ENV=development

GITHUB_CLIENT_ID=abcd
GITHUB_CLIENT_SECRET=efgh

DISCORD_CLIENT_ID=hijk
DISCORD_CLIENT_SECRET=lmno
```

### CODE REVIEW

File: server/controller/passportControllers.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                        accessToken,
+                        refreshToken
+                    }
+                };
+                const updatedUser = await User.findOneAndUpdate(
+                    {
+                        accounts: { $elemMatch: { kind: 'github', 'token.accessToken': token } }
+                    },
+                    { discordProfile: profile, $push: { accounts: discordTokens } },
+                    { new: true }
```
Comment (darshkpatel) [inline server/controller/passportControllers.js:94]: do we really need this ?

File: server/env.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+dotenv.config();
+
+export const {
+    DB_NAME,
+    MONGO_PASSWORD,
+    API_KEY,
+    SALT_ROUNDS
+} = process.env;
+
+// TODO: Replace with Atlas URL
```
Comment (darshkpatel) [inline server/env.js]: `docker run --rm -v mongo_data:/data/db -p 27017:27017 mongo` FTW

**Review by darshkpatel — APPROVED** [review]

**Review by kunal-kushwaha — APPROVED** [review]

**Comment by kunal-kushwaha** [comment]
Are we not adding Google auth?

**Comment by darshkpatel** [comment]
Google Auth is needed for calendar events, which can be put on low priority

File: server/controller/passportControllers.js (hunk 1 above)
Comment (sauravhiremath) [reply server/controller/passportControllers.js:94]: For reading the channel messages for search feature. yes

Could you specify what this means?

File: server/env.js (hunk 2 above)
Comment (sauravhiremath) [reply server/env.js]: yeah, deploying it on vps. So no need of atlas!

File: server/controller/passportControllers.js (hunk 1 above)
Comment (sauravhiremath) [reply server/controller/passportControllers.js:94]: Yeah, `{ new: true }` is required for returning updated user (without discord data added to github)
Without it, `findOneAndUpdate()` returns the old document

**Pull request merged**